src/course-outline/data/api.ts

```diff
@@ -382,19 +382,38 @@ export async function duplicateCourseItem(itemId: string, parentId: string): Pro
 }
 
 /**
- * Add new course item like section, subsection or unit.
- * @param {string} parentLocator
- * @param {string} category
- * @param {string} displayName
- * @returns {Promise<Object>}
+ * Creates a new course XBlock. Can be used to create any type of block
+ * and also import a content from library.
  */
-export async function addNewCourseItem(parentLocator: string, category: string, displayName: string): Promise<object> {
+export async function createCourseXblock({
+  type,
+  category,
+  parentLocator,
+  displayName,
+  boilerplate,
+  stagedContent,
+  libraryContentKey,
+}: {
+  type: string,
+  category?: string, // The category of the XBlock. Defaults to the type if not provided.
```

Comment:
nit: that way the docs work
```suggestion
  /** The category of the XBlock. Defaults to the type if not provided. */
  category?: string,
```